lib/simplecov/configuration.rb

```diff
@@ -456,6 +469,12 @@ def minimum_possible_coverage_exceeded(coverage_option)
       warn "The coverage you set for #{coverage_option} is greater than 100%"
     end
 
+    def order_default_first(coverage)
+      {
+        default_coverage_criterion => coverage[default_coverage_criterion]
+      }.merge(coverage).compact
```

Comment:
This I'm not too sure about. I know why it works but call me old school, I'm not a fan of treating hashes as if they have an order (although they do these days in Ruby). That we display the "primary" coverage first is actually not that important to me so personally I'd be ok with us leaving the order up to randomness/the order in which users provide it. 

That said, I'm not set here but I think we could avoid this little bit of complexity :)

Reply:
I like the code a lot better without the ordering too, so will remove it!